src/Fixer/Whitespace/NoExtraWhitespaceFixer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Remove extra whitespace buried inside of any SLOC.
+ *
+ * @author Tristan Strathearn <r3oath@gmail.com>
+ */
+final class NoExtraWhitespaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if ($token->isWhitespace(" \0\x0B")) {
+                $space = $token->getContent();
+
+                // Here we'll normalize all consecutive whitespace into a
+                // single whitespace character.
+                if (strlen($space) > 1) {
+                    $token->setContent(' ');
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Remove extra whitespace buried inside of any SLOC.';
+    }
+
+    public function getPriority()
+    {
+        // Make sure this is generally run before any other whitespace fixer.
```

Comment:
As I eluded too in my previous comment, what would be considered a reliable test then? As it's currently set to come before any whitespace fixer... am I listing the order of this fixer then every other whitespace fixer currently available? or just choosing some random whitespace fixer I know the priority of and making sure it comes before that. Just seems this system doesn't make much sense unless you're writing something that very specifically had to be run before or after a limited set of fixers.